contracts/Staking.sol

```diff
@@ -316,14 +333,20 @@ contract Staking is Ownable, ReentrancyGuard {
     ) internal {
         for (uint256 index = 0; index < allowedTokens.length; index++) {
             address tokenAddress = allowedTokens[index];
-            uint256 usersStake = usersPool[tokenAddress];
 
-            uint256 compensation = usersStake
+            uint256 compensation = usersPool[tokenAddress]
                 .mul(leverage)
                 .mul(_rewardPercentage)
                 .div(_precision);
 
-            providerPool[tokenAddress] = providerPool[tokenAddress].sub(
+            // Compensation must be less than 25% of providersPool to ensure payout at all time
+            if (compensation > providersPool[tokenAddress].mul(25).div(100)) {
+                compensation = providersPool[tokenAddress]
```

Comment:
Same interrogation, why not setting compensation at `providersPool[tokenAddress].mul(25).div(100)` to prevent instability?

Reply:
See my other answer, that's also because _rewardPercentage is already capped at 25% is the SLORegistry

Here the compensation is `25% * leverage * usersPool` if there is enough liquidity in providersPool, or simply **up to** 25% of the providersPool